File: backend/tests/fixtures/security_test_repo/src/vulnerable.js
/**
 * SECURITY TEST FIXTURE — JavaScript
 * Intentionally vulnerable examples for Phase 6 testing.
 * FAKE test credentials only — do not use in production.
 */

// FAKE test token — synthetic only
const ACCESS_TOKEN = "TEST_SECRET_DO_NOT_USE_ghp_fake12345678901234567890";

// XSS via dangerouslySetInnerHTML
function UserGreeting({ name }) {
  return (
    <div dangerouslySetInnerHTML={{ __html: name }} />
  );
}

// console with sensitive data
function login(password) {
  console.log("Login attempt with password:", password);
  return authenticate(password);
}

// Hardcoded localhost URL
const API_URL = "http://localhost:3000/api";

// SSRF-like pattern
async function fetchUserData(userId) {
  const url = req.query.url;  // user controlled
  const response = await fetch(url);
  return response.json();
}

// eval usage
function calculate(expr) {
  return eval(expr);
}

// TODO marker
// TODO: add input validation here
function processInput(input) {
  return input;
}

File: backend/tests/fixtures/security_test_repo/src/vulnerable.py
"""
SECURITY TEST FIXTURE — Python
Intentionally vulnerable examples for DevPilot Phase 6 testing.
These are SYNTHETIC vulnerabilities. Do not use this in production.
All secrets here are FAKE test credentials only.
"""

import subprocess
import os

# FAKE test credentials — clearly synthetic, not real
# TEST_SECRET_DO_NOT_USE — these are fake values for scanner testing only
API_KEY = "TEST_SECRET_DO_NOT_USE_123456"
DATABASE_PASSWORD = "fake_test_password_abcdefgh"


# SQL injection — user input directly concatenated
def get_user(db, user_id):
    # This is intentionally unsafe for testing
    query = "SELECT * FROM users WHERE id = " + user_id
    return db.execute(query)


# Command injection — user input in subprocess
def run_command(user_input):
    # This is intentionally unsafe for testing
    result = subprocess.run(["ls", user_input], capture_output=True)
    return result


# Hardcoded secret in function (different pattern)
def connect_db():
    # FAKE: synthetic test value
    connection_string = "postgres://admin:FAKE_TEST_PASS_9999@localhost:5432/mydb"
    return connection_string


# Broad exception handling
def risky_operation():
    try:
        result = eval("1 + 1")  # eval without user input
        return result
    except Exception:
        pass


# TODO marker
def unfinished_work():
    # TODO: implement proper validation
    pass
